scaleway/resource_instance_security_group_rules.go

```diff
@@ -0,0 +1,138 @@
+package scaleway
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"strings"
+)
+
+func resourceScalewayInstanceSecurityGroupRules() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayInstanceSecurityGroupRulesCreate,
+		Read:   resourceScalewayInstanceSecurityGroupRulesRead,
+		Update: resourceScalewayInstanceSecurityGroupRulesUpdate,
+		Delete: resourceScalewayInstanceSecurityGroupRulesDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"security_group_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The security group associated with this volume",
+			},
+			"inbound_rule": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Inbound rules for this set of security group rules",
+				Elem:        securityGroupRuleSchema(),
+			},
+			"outbound_rule": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Outbound rules for this set of security group rules",
+				Elem:        securityGroupRuleSchema(),
+			},
+			"zone":            zoneSchema(),
+			"organization_id": organizationIDSchema(),
+		},
+	}
+}
+
+func resourceScalewayInstanceSecurityGroupRulesCreate(d *schema.ResourceData, m interface{}) error {
+	d.SetId(securityGroupRulesIDFromSecurityGroupID(d))
+
+	// We call update instead of read as it will take care of creating rules.
+	return resourceScalewayInstanceSecurityGroupUpdate(d, m)
+}
+
+func resourceScalewayInstanceSecurityGroupRulesRead(d *schema.ResourceData, m interface{}) error {
+
+	securityGroupRulesZoneID := d.Id()
+	securityGroupZoneID := securityGroupZIDFromsecurityGroupRulesZID(securityGroupRulesZoneID)
```

Comment:
keep variable naming consistent with other resources

Reply:
I find using `ID`when it is in fact a `zonedID`confusing. 

Also, this is a special case where whe have 2 `IDs` because one is built from the other. Just using `ID` would be confusing.